app/services/send_notification.rb

```diff
@@ -10,7 +10,7 @@ def self.call(...)
   end
 
   def execute
-    return unless @account.notification?
+    return unless @account.notification? && @account.email.present?
```

Comment:
Would not it be easier just to add a validation on the model that prevents savings the checked mark without email?

Reply:
1) It limit a user + we add an additional button/action for the user. In the app we haven't had any validation messages for user yet. Can we stop without messages?
I picked it up from our boss who said [here](https://widefixworkspace.slack.com/archives/D05MFV0QHB8/p1696018971652719?thread_ts=1696016328.813389&cid=D05MFV0QHB8): the fewer buttons are now the easier it will be to keep a user. It means no resistance. 

2) `@account.email.present?` was added for two account with the checked box without emails to abstain the errors. I can toggle the boxes, with your permissions. And remove the method/conditional

Generally you right and I'm on the same page(!): stop user with validation + the message.

Reply:
WDYT if we fix the existing invalid data in prod after adding the validation on the model? Would it solve the 1st concern?

Reply:
Unfortunately not.
As you see, here is two issues: stop the user with validation + messages the prod with the errors and.
The last one could be resolved effortlessly!
The first one is too complicated:
  1) We shouldn't stop users at all, at least for two-three months. 
  2) After that has better create validations + messages + (!)child email verification(i'll add the task on the board)

Reply:
Also please take into account the number of accounts with invalid state (if the new validation added) in prod.

Reply:
`Stop` means if the user checked the box 'Report transactions' with the empty email it don't pass the user with saving the account. We have to add messages. It take time for the user.

Reply:
I would do this:
1. add validation to the model that does not allow saving the checkbox selected if no email is provided
2. check production data and fix it (after the fix release)
3. now I consider that the prod data is consistent and valid, so do I need this `@account.email.present?` check
4. also I don't need any custom code in the controller

Are there any issues with this plan? Why it might not work?

Reply:
Got it, thanks!

Reply:
1. done
2. fixed prod data before the release
3. now, the app don't have accounts with notification without emails and its won't be with the validation, hence remove the `@account.email.present?`
4. done
Thx!

Reply:
Thanks! Note, we need to repeat 2 after release, agreed?

Reply:
Agreed.